lib/functions.php

```diff
@@ -214,14 +214,16 @@ function wait($promise)
         $resolved = false;
 
         try {
-            Loop::run(function () use (&$resolved, &$value, &$exception, $promise) {
-                $promise->onResolve(function ($e, $v) use (&$resolved, &$value, &$exception) {
-                    Loop::stop();
-                    $resolved = true;
-                    $exception = $e;
-                    $value = $v;
+            do {
+                Loop::run(function () use (&$resolved, &$value, &$exception, $promise) {
+                    $promise->onResolve(function ($e, $v) use (&$resolved, &$value, &$exception) {
+                        Loop::stop();
+                        $resolved = true;
+                        $exception = $e;
+                        $value = $v;
+                    });
                 });
-            });
+            } while (!$resolved && Loop::getInfo()['enabled_watchers']['referenced'] > 0);
```

Comment:
Since the promise is resolved, `Loop::stop()` will be called on every tick of the event loop.

Reply:
Fixed.